Count Frameworld Decompresspr/Penrose/PCC towards achievements

PolarisWasInDenial opened the pull request:
- Adds a check for the deposit created by the Frameworld computing upgrade for the Megastructure Mogul achievement
- Adds a check for the Integrated Matter Decompressor megastructure of the Frameworld for the Black Hole Acquisition achievement chain
- Adds a check for the Integrated Penrose Sphere to count towards Megastructure achievements at the same tier as Stabilized Penrose

PolarisWasInDenial committed:
Count Frameworld Decomp/Penrose/PCC towards achievements

- Adds a check for the deposit created by the frameworld computing upgrade for the Megastructure Mogul achievement
- Adds a check for the Integrated Matter Decompressor megastructure of the Frameworld for the Black Hole Acquisition achievement chain
- Adds a check for the Integrated Penrose Sphere to count towards Megastructure achievements at the same tier as Stabilized Penrose

PolarisWasInDenial commented:
Running a test locally before marking this as ready for review.

PolarisWasInDenial commented:
Appears to work locally with changes run against current release build, marking Ready for Review.

Bread-TM commented:
> Appears to work locally with changes run against current release build, marking Ready for Review.

So to be sure, you're saying it causes no issues with the achievement changes on live and you've tested and made sure of that correct?

Bread-TM committed:
Made the check slightly more performant

- Its better to check for it being a frameworld first so that it doesn't have to check all the deposits, also you don't want to run any_planet_within_borders twice.
